### ISSUE TITLE
Unexpected error: Cannot read property 'totalFoundErrorNumber' of undefined

### ISSUE THREAD

**Issue by wheelerlaw** [nvuillam/npm-groovy-lint#155]
I am getting an error when running the following command:

```shell
npm-groovy-lint --noserver --failon warning --path ./ --output ./linting.txt --loglevel warning 
```

Here is my configuration:

```json
{
  "extends": "recommended",
  "rules": {
    "BuilderMethodWithSideEffects": "off",
    "CatchException": "off",
    "CompileStatic": "off",
    "DuplicateListLiteral": "off",
    "DuplicateMapLiteral": "off",
    "DuplicateNumberLiteral": "off",
    "DuplicateStringLiteral": "off",
    "FactoryMethodName": "off",
    "GStringAsMapKey": "off",
    "GStringExpressionWithinString": "off",
    "GrailsMassAssignment": "off",
    "ImplicitClosureParameter": "off",
    "Instanceof": "off",
    "JUnitPublicNonTestMethod": "off",
    "JUnitPublicProperty": "off",
    "JavaIoPackageAccess": "off",
    "LineLength": "off",
    "LoggerWithWrongModifiers": "off",
    "MethodCount": "off",
    "MethodName": "off",
    "NestedBlockDepth": "off",
    "NoFloat": "off",
    "NonFinalPublicField": "off",
    "ParameterCount": "off",
    "PrintStackTrace": "off",
    "PublicInstanceField": "off",
    "ReturnNullFromCatchBlock": "off",
    "ThrowError": "off",
    "UnnecessaryGString": "off",
    "UnnecessaryToString": "off",
    "Indentation": {
      "spacesPerIndentLevel": 4,
      "severity": "warning"
    },

    "AssignmentInConditional": "warning",
    "BlankLineBeforePackage": "warning",
    "BlockEndsWithBlankLine": "warning",
    "BlockStartsWithBlankLine": "warning",
    "BracesForClass": "warning",
    "BracesForForLoop": "warning",
    "BracesForIfElse": "warning",
    "BracesForMethod": "warning",
    "BracesForTryCatchFinally": "warning",
    "ClassEndsWithBlankLine": "warning",
    "ClassStartsWithBlankLine": "warning",
    "ClosingBraceNotAlone": "warning",
    "ConsecutiveBlankLines": "warning",
    "DuplicateImport": "warning",
    "ElseBlockBraces": "warning",
    "ExplicitArrayListInstantiation": "warning",
    "ExplicitLinkedListInstantiation": "warning",
    "FileEndsWithoutNewline": "warning",
    "IfStatementBraces": "warning",
    "IndentationClosingBraces": "warning",
    "IndentationComments": "warning",
    "InsecureRandom": "warning",
    "MisorderedStaticImports": "warning",
    "MissingBlankLineAfterImports": "warning",
    "MissingBlankLineAfterPackage": "warning",
    "NoTabCharacter": "warning",
    "SpaceAfterCatch": "warning",
    "SpaceAfterComma": "warning",
    "SpaceAfterFor": "warning",
    "SpaceAfterIf": "warning",
    "SpaceAfterOpeningBrace": "warning",
    "SpaceAfterSemicolon": "warning",
    "SpaceAfterSwitch": "warning",
    "SpaceAfterWhile": "warning",
    "SpaceAroundOperator": "warning",
    "SpaceBeforeClosingBrace": "warning",
    "SpaceBeforeOpeningBrace": "warning",
    "TrailingWhitespace": "warning",
    "UnnecessaryDefInFieldDeclaration": "warning",
    "UnnecessaryDefInMethodDeclaration": "warning",
    "UnnecessaryDefInVariableDeclaration": "warning",
    "UnnecessaryDotClass": "warning",
    "UnnecessaryFinalOnPrivateMethod": "warning",
    "UnnecessaryGroovyImport": "warning",
    "UnnecessaryInstantiationToGetClass": "warning",
    "UnnecessaryPackageReference": "warning",
    "UnnecessaryParenthesesForMethodCallWithClosure": "warning",
    "UnnecessarySemicolon": "warning",
    "UnusedImport": "warning"
  }
}
```

If I run `npm-groovy-lint`, I get an error:

```
$ npm-groovy-lint --noserver --failon warning --path ./ --output ./linting.txt --loglevel warning 
GroovyLint: Error running CodeNarc: 
java.lang.NullPointerException: Cannot get property 'name' on null object
        at org.codehaus.groovy.runtime.NullObject.getProperty(NullObject.java:60)
        at org.codehaus.groovy.runtime.InvokerHelper.getProperty(InvokerHelper.java:194)
        at org.codehaus.groovy.runtime.callsite.NullCallSite.getProperty(NullCallSite.java:46)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGetProperty(AbstractCallSite.java:329)
        at org.codenarc.ruleset.JsonReaderRuleSet$_loadRuleElements_closure1.doCall(JsonReaderRuleSet.groovy:64)
        at sun.reflect.GeneratedMethodAccessor3.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:263)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1026)
        at groovy.lang.Closure.call(Closure.java:412)
        at org.codehaus.groovy.runtime.DefaultGroovyMethods.callClosureForMapEntry(DefaultGroovyMethods.java:5947)
        at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2395)
        at org.codehaus.groovy.runtime.dgm$201.invoke(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:247)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
        at org.codenarc.ruleset.JsonReaderRuleSet.loadRuleElements(JsonReaderRuleSet.groovy:60)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:61)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:171)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:185)
        at org.codenarc.ruleset.JsonReaderRuleSet.<init>(JsonReaderRuleSet.groovy:43)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:72)
        at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:105)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:59)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:263)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:277)
        at org.codenarc.ruleset.RuleSetUtil.loadRuleSetFromString(RuleSetUtil.groovy:42)
        at org.codenarc.ruleset.RuleSetUtil$loadRuleSetFromString.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
        at org.codenarc.CodeNarcRunner.createInitialRuleSetFromString(CodeNarcRunner.groovy:173)
        at org.codenarc.CodeNarcRunner.createInitialRuleSet(CodeNarcRunner.groovy:168)
        at org.codenarc.CodeNarcRunner.buildRuleSet(CodeNarcRunner.groovy:132)
        at org.codenarc.CodeNarcRunner.execute(CodeNarcRunner.groovy:87)
        at org.codenarc.CodeNarcRunner$execute.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:130)
        at org.codenarc.CodeNarc.execute(CodeNarc.groovy:186)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:193)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:73)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
        at org.codenarc.CodeNarc.main(CodeNarc.groovy:156)

Unexpected error: Cannot read property 'totalFoundErrorNumber' of undefined
TypeError: Cannot read property 'totalFoundErrorNumber' of undefined
    at NpmGroovyLint.manageReturnCode (/home/wlaw/.npm/lib/node_modules/npm-groovy-lint/lib/groovy-lint.js:481:49)
    at NpmGroovyLint.postProcess (/home/wlaw/.npm/lib/node_modules/npm-groovy-lint/lib/groovy-lint.js:330:14)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async NpmGroovyLint.run (/home/wlaw/.npm/lib/node_modules/npm-groovy-lint/lib/groovy-lint.js:68:13)
    at async /home/wlaw/.npm/lib/node_modules/npm-groovy-lint/lib/index.js:12:9
```

If I shorten the config file to this, it works:

```json
{
  "extends": "recommended",
  "rules": {
    "BuilderMethodWithSideEffects": "off",
    "CatchException": "off",
    "CompileStatic": "off",
    "DuplicateListLiteral": "off",
    "DuplicateMapLiteral": "off",
    "DuplicateNumberLiteral": "off",
    "DuplicateStringLiteral": "off",
    "FactoryMethodName": "off",
    "GStringAsMapKey": "off",
    "GStringExpressionWithinString": "off",
    "GrailsMassAssignment": "off",
    "ImplicitClosureParameter": "off",
    "Instanceof": "off",
    "JUnitPublicNonTestMethod": "off",
    "JUnitPublicProperty": "off",
    "JavaIoPackageAccess": "off",
    "LineLength": "off",
    "LoggerWithWrongModifiers": "off",
    "MethodCount": "off",
    "MethodName": "off",
    "NestedBlockDepth": "off",
    "NoFloat": "off",
    "NonFinalPublicField": "off",
    "ParameterCount": "off",
    "PrintStackTrace": "off",
    "PublicInstanceField": "off",
    "ReturnNullFromCatchBlock": "off",
    "ThrowError": "off",
    "UnnecessaryGString": "off",
    "UnnecessaryToString": "off",
    "Indentation": {
      "spacesPerIndentLevel": 4,
      "severity": "warning"
    },

    "AssignmentInConditional": "warning",
    "BlankLineBeforePackage": "warning",
    "BlockEndsWithBlankLine": "warning",
    "BlockStartsWithBlankLine": "warning",
    "BracesForClass": "warning",
    "BracesForForLoop": "warning",
    "BracesForIfElse": "warning",
    "BracesForMethod": "warning",
    "BracesForTryCatchFinally": "warning",
    "ClassEndsWithBlankLine": "warning",
    "ClassStartsWithBlankLine": "warning"
  }
}
```

Basically I want all auto-fixable rules to throw a warning (so that they would fail a CI run). But it seems like there are issues when I set all of them to warning.

**Comment by wheelerlaw** [comment]
So it seems that the offending rule is `ClosingBraceNotAlone`, CodeNarc does not support that rule:

This function throws a NPE since the string variable `ruleName` will be `ClosingBraceNotAlone`, which `RuleRegistryHolder.ruleRegistry?.getRuleClass(ruleName)` will return null:

```groovy
    private void loadRuleElements(Map<String,Map> ruleset) {
        ruleset.each { String ruleName, Map ruleParams ->
            // ruleName can be a CodeNarc class or just RuleName
            def ruleNameSplit = ruleName.tokenize('.')
            def ruleClassName = ruleNameSplit.size() > 2 ? ruleName : // ex: org.codenarc.rule.exception.CatchThrowableRule
                            RuleRegistryHolder.ruleRegistry?.getRuleClass(ruleName).name // ex: CatchThrowableRule
            assert ruleClassName != 'null', 'Unable to identify class from string ' + ruleName
            def ruleClass = getClass().classLoader.loadClass(ruleClassName)
            RuleSetUtil.assertClassImplementsRuleInterface(ruleClass)
            def rule = ruleClass.newInstance()
            rules << rule
            setRuleProperties(ruleParams, rule)
        }
    }
```

**Comment by wheelerlaw** [comment]
Alright, so it turns out that `ClosingBraceNotAlone`, `IndentationClosingBraces`, and  `IndentationComments` are invalid rules.
